src/stores/right-panel/RightPanelStoreIPanelState.ts

```diff
@@ -0,0 +1,127 @@
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
+    // group
+    groupId?: string;
+    groupRoomId?: string;
+    widgetId?: string;
+    spaceId?: string;
+    // Room3pidMemberInfo, Space3pidMemberInfo,
+    memberInfoEvent?: MatrixEvent;
+    // threads
+    threadHeadEvent?: MatrixEvent;
+    initialEvent?: MatrixEvent;
+    isInitialEventHighlighted?: boolean;
+}
+
+export interface IRightPanelCardStateStored {
+    member?: RoomMember | User | GroupMember;
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
+    // group
+    groupId?: string;
+    groupRoomId?: string;
+    widgetId?: string;
+    spaceId?: string;
+    // 3pidMemberInfo
+    memberInfoEventId?: string;
+    // threads
+    threadHeadEventId?: string;
+    initialEventId?: string;
+    isInitialEventHighlighted?: boolean;
+}
+
+export interface IRightPanelCard {
+    phase: RightPanelPhases;
+    state?: IRightPanelCardState;
+}
+
+export interface IRightPanelCardStored {
+    phase: RightPanelPhases;
+    state?: IRightPanelCardStateStored;
+}
+
+export interface IRightPanelForRoom {
+    isOpen: boolean;
+    history: Array<IRightPanelCard>;
+}
+
+interface IRightPanelForRoomStored {
+    isOpen: boolean;
+    history: Array<IRightPanelCardStored>;
+}
+
+export function convertToStorePanel(cacheRoom: IRightPanelForRoom): IRightPanelForRoomStored {
+    if (!cacheRoom) return cacheRoom;
+    const storeHistory = [...cacheRoom.history].map(panelState => convertCardToStore(panelState));
+    return { isOpen: cacheRoom.isOpen, history: storeHistory };
+}
+
+export function convertToStatePanel(storeRoom: IRightPanelForRoomStored, room: Room): IRightPanelForRoom {
+    if (!storeRoom) return storeRoom;
+    const stateHistory = [...storeRoom.history].map(panelStateStore => convertStoreToCard(panelStateStore, room));
+    return { history: stateHistory, isOpen: storeRoom.isOpen };
+}
+
+export function convertCardToStore(panelState: IRightPanelCard): IRightPanelCardStored {
+    const panelStateThisRoomStored = { ...panelState.state } as any;
+    if (!!panelState?.state?.threadHeadEvent?.getId()) {
+        panelStateThisRoomStored.threadHeadEventId = panelState.state.threadHeadEvent.getId();
+    }
+    if (!!panelState?.state?.memberInfoEvent?.getId()) {
+        panelStateThisRoomStored.memberInfoEventId = panelState.state.memberInfoEvent.getId();
+    }
+    if (!!panelState?.state?.initialEvent?.getId()) {
+        panelStateThisRoomStored.initialEventId = panelState.state.initialEvent.getId();
+    }
+    delete panelStateThisRoomStored.threadHeadEvent;
+    delete panelStateThisRoomStored.initialEvent;
+    delete panelStateThisRoomStored.memberInfoEvent;
+
+    const storedCard = { state: panelStateThisRoomStored as IRightPanelCardStored, phase: panelState.phase };
+    return storedCard as IRightPanelCardStored;
```

Comment:
Why are all these type assertions (`as`) needed in these conversion functions...?

Reply:
Oh... I did not pay attention and did this change in the other PR: [`3c7068c` (#7398)](https://github.com/matrix-org/matrix-react-sdk/pull/7398/commits/3c7068ca3112c04ebf561500f3078d06950b35ad)

Reply:
I think we can keep it like this. since the two prs can be merged almost simultaneously now